### PR TITLE
docs(docs): Add API Versioning ADR

### PR DESCRIPTION
**Pull request by stevenj**
# Description

This adds an ADR to describe our API versioning strategy and rules.

## Description of Changes

Documentation only to add the new ADR.

Rendered Version: https://input-output-hk.github.io/catalyst-voices/branch/adr_api_endpoints/architecture/09_architecture_decisions/0007-api-versions/

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

**Review by nathanbogale — DISMISSED** [review]

**Review by neil-iohk — DISMISSED** [review]

File: docs/src/architecture/09_architecture_decisions/0007-api-versions.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+All Individual API Endpoints will be named and versioned according to the following rules:
+
+1. All endpoints URIs will be under the `/api` path in the backend.
+2. The general structure of any endpoint will be `/api/<version>/<path>`
+3. `<version>` defines a version of the endpoint supplied by a `<path>` and is **NOT** aligned with any other versions.
+4. `<version>` **MUST** either be `draft` or matching the regular expression `^v(?!0)\d{1,}$`[^1]
+5. `<path>` Can be any path of at least one element as required for the endpoint.
+6. **ALL** new endpoints start as `draft` they are promoted to a versioned endpoint by a deliberate decision.
+7. New endpoints **MUST NOT** be versioned other than `draft` in their initial PR.
```
Comment (minikin) [inline docs/src/architecture/09_architecture_decisions/0007-api-versions.md:36]: Can we please consider adding an example of such a `draft` endpoint as we did above `/api/<version>/<path>`?

File: docs/src/architecture/09_architecture_decisions/0007-api-versions.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+in the schema definition where the additional field is to be added.
+    * A previously undocumented behavior is documented.  
+In this case, the behavior must already exist, it is not a breaking change to clarify documentation.
+19. *ALL* non-breaking changes proposed in a PR to existing API's,
+or migrating an API from `draft` to a versioned API **MUST** be signed off by
+    * at least 1 Architect on the Team; and
+    * the Engineering manager; and preferably
+    * A senior member of the team with primary responsibility for the frontend.
+20. When a new version of an endpoint is released, the existing endpoint **MUST** be marked as `deprecated`
+in the OpenAPI specification for the endpoint.
```
Comment (minikin) [inline docs/src/architecture/09_architecture_decisions/0007-api-versions.md:63]: I find it difficult to understand how this should work. Could you please add an example?

**Review by minikin — CHANGES_REQUESTED** [review]
That's a beneficial ADR, and it looks good to me overall.

File: docs/src/architecture/09_architecture_decisions/0007-api-versions.md (hunk 1 above)
Comment (stevenj) [reply docs/src/architecture/09_architecture_decisions/0007-api-versions.md:36]: Examples added

File: docs/src/architecture/09_architecture_decisions/0007-api-versions.md (hunk 2 above)
Comment (stevenj) [reply docs/src/architecture/09_architecture_decisions/0007-api-versions.md:63]: I added a link to how OpenAPI manages deprecation.
When we formalize deprecation, we would need another ADR which explains it.
There are a few options we can consider in order to document it, and advise consumers of our API about whats happened.

**Review by minikin — APPROVED** [review]
LGTM

**Pull request merged**